src/remoted/manager.c

```diff
@@ -380,8 +384,36 @@ void c_group(const char *group, char ** files, file_sum ***_f_sum,char * sharedc
             os_calloc(1, sizeof(file_sum), f_sum[f_size]);
             strncpy(f_sum[f_size]->sum, md5sum, 32);
             os_strdup(files[i], f_sum[f_size]->name);
-
-            if (!logr.nocmerged) {
+            
+            if(modified = (struct tm *) OSHash_Get(invalid_files,file), modified){
+                struct stat attrib;
+                struct tm *last_modify;
+
+                stat(file, &attrib);
+                last_modify = gmtime(&(attrib.st_mtime));
+                if( modified != last_modify){
+                    if(checkBinaryFile(file)){
+                        OSHash_Set(invalid_files, file, last_modify);
+                        ignored = 1;
+                    }
+                    else{
+                        OSHash_Delete(invalid_files, file);
+                    }
+                }
+            }
+            else {
+                if(checkBinaryFile(file)){
+                    struct stat attrib;
+                    struct tm *last_modify;
+
+                    stat(file, &attrib);
+                    last_modify = gmtime(&(attrib.st_mtime));
+                    OSHash_Add(invalid_files, file, last_modify);
+                    ignored = 1;
+                    merror("File %s was detected as an invalid file",file);
```

Comment:
I think the error message could be more descriptive: `Invalid shared file X in group Y. Ignoring it.`